lib/kube/proxy/forwarder.go

```diff
@@ -728,8 +772,35 @@ func (f *Forwarder) getKubeAccessDetails(
 
 		// Get list of allowed kube user/groups based on kubernetes service labels.
 		labels := types.CombineLabels(c.GetStaticLabels(), types.LabelsToV2(c.GetDynamicLabels()))
-		labelsMatcher := services.NewKubernetesClusterLabelMatcher(labels)
-		groups, users, err := roles.CheckKubeGroupsAndUsers(sessionTTL, false, labelsMatcher)
+
+		matchers := make([]services.RoleMatcher, 1, 2)
+		// Creates a matcher that matches the cluster labels against `kubernetes_labels`
+		// defined for each user's role.
+		matchers[0] = services.NewKubernetesClusterLabelMatcher(labels)
```

Comment:
Suggestion: the combination of using a fixed length and using appends is a bit unusual. Use length=0 and always append?

```suggestion
		matchers := make([]services.RoleMatcher, 0, 2)
		// Creates a matcher that matches the cluster labels against `kubernetes_labels`
		// defined for each user's role.
		matchers = append(matchers, services.NewKubernetesClusterLabelMatcher(labels))
```